dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/icon/jdbc/JdbcCustomIconStore.java

```diff
@@ -87,15 +88,19 @@ public List<CustomIcon> getIconsByKeywords(String[] keywords) {
   }
 
   @Override
-  public List<CustomIcon> getAllIcons() {
+  public List<CustomIcon> getAllIcons(IconCriteria iconCriteria) {
     final String sql =
         """
             select c.key as iconkey, c.description as icondescription, c.keywords as keywords, f.uid as fileresourceuid, u.uid as useruid
             from customicon c join fileresource f on f.fileresourceid = c.fileresourceid
             join userinfo u on u.userinfoid = c.createdby
             """;
 
-    return jdbcTemplate.query(sql, customIconRowMapper);
+    if (Boolean.TRUE.equals(iconCriteria.isSkipPaging())) {
```

Comment:
can this be only `if (iconCriteria.isSkipPaging()) {`?

Reply:
@muilpp Yes, it can be. But Sonarcloud does not like it and complains :)

Reply:
in this case, I think we should use `if (iconCriteria.isPagingRequest())`
https://github.com/dhis2/dhis2-core/blob/master/dhis-2/dhis-api/src/main/java/org/hisp/dhis/webapi/controller/event/webrequest/PagingAndSortingCriteriaAdapter.java#L70